File: ruby/transaction.rb
class Transaction
  attr_reader :name, :category, :amount

  def initialize(name:, category:, amount:)
    @name = name
    @category = category
    @amount = amount
  end

  def to_s
    "Transaction for $#{@amount}: [#{@category}] #{@name}"
  end
end

File: javascript/transaction.js
class Transaction {
  constructor(name, category, amount) {
    this.name = name;
    this.category = category;
    this.amount = amount;
  }

  toString() {
    return `Transaction for $${this.amount}: [${this.category}] ${this.name}`;
  }
}

export default Transaction;
